### ISSUE TITLE
Method A-D, I | test case

### ISSUE THREAD

**Issue by chraibi** [JuPedSim/jpsreport#130]
In Gitlab by @gjaeger on Mar 7, 2019, 08:53 [[origin](https://gitlab.version.fz-juelich.de/jupedsim/jpsreport/issues/102)]

test case for method A-D and I

**Comment by chraibi** [comment]
In Gitlab by @gjaeger on Mar 23, 2019, 10:22

notes for a quick reminder:
- issue #9 - **Make Utest** - CDash like 
- issue #11 - **Add test FD** - Compare the FD of jpsreport with a reference result file
- issue #58 - **Write unit test** - Write a python script that reproduces Figures **5** and **6** in [Transitions in pedestrian fundamental diagrams of straight corridors and T-junctions](http://juser.fz-juelich.de/record/16126).
- issue #65 - **check method C** - too slow ...

**Comment by chraibi** [comment]
In Gitlab by @chraibi on Mar 23, 2019, 16:04

CDash is a system we don't use anymore. It has been replaced by Gitlab's CI.

For the rest, I would suggest to make a directory for each of them:

```
Utest
├── test1
├── test2
└── test3
```

**Comment by chraibi** [comment]
In Gitlab by @gjaeger on Mar 25, 2019, 08:21

notes (for me):

*gitlab_ci.yml*: 
- [GitLab CI/CD Pipeline Configuration Reference](https://docs.gitlab.com/ee/ci/yaml/)

example:
```yml
#============== unit_test ============
test:
  dependencies:
    - make-linux

  stage: unit_test
  script:
    - cd build
    - ctest -R Boost
    - ctest -R clean
  tags:
    - linux

```

*structure*: 
- see folder [Utest](https://gitlab.version.fz-juelich.de/jupedsim/jpscore/tree/develop/Utest) in jpscore

**Comment by chraibi** [comment]
In Gitlab by @gjaeger on Mar 25, 2019, 08:21

changed due date to March 28, 2019

**Comment by chraibi** [comment]
In Gitlab by @gjaeger on Apr 1, 2019, 09:48

changed due date to April 05, 2019

**Comment by chraibi** [comment]
In Gitlab by @gjaeger on Jun 5, 2019, 10:25

The geometry: [geo_Uebung_02.xml](https://gitlab.version.fz-juelich.de/JuPedSim/jpsreport/uploads/6b92fdc8e2a3e90a66a31f91ef634df2/geo_Uebung_02.xml)  

![Raum_02b_oc](/uploads/fbaebb128e14ea578bbf36de02518d50/Raum_02b_oc.png)

The trajectory file: [traj_Uebung_02b.txt](/uploads/d9132ff4db77b58df1bdd01caf4c428b/traj_Uebung_02b.txt)

*measurement area for method A*   
![Raum_02_MA_A](/uploads/6cbcbb96152782e47c4f84f7ea9fc155/Raum_02_MA_A.png)

*measurement area for method B*   
![Raum_02_MA_B](/uploads/ff028450778bb1de5a8bcb76237f6a4e/Raum_02_MA_B.png)     

*measurement area for method C and D*      
![Raum_02_MA_C-D](/uploads/ea6b5c92afd578d133d51118f94a3408/Raum_02_MA_C-D.png)

**Comment by chraibi** [comment]
In Gitlab by @gjaeger on Jun 5, 2019, 10:47

WIP: new branch [102-Utest](https://gitlab.version.fz-juelich.de/jupedsim/jpsreport/tree/102-Utest)

**Comment by gjaeger** [comment]
see PR #152

**Comment by chraibi** [comment]
This issue needs to be closed immediately, otherwise we can not test any upcoming PR.

**Comment by gjaeger** [comment]
solved with PR #152
